### ISSUE TITLE
Can `build.hxml` be looked recursively in parent dirs?

### ISSUE THREAD

**Issue by emugel** [vshaxe/vshaxe#412]
Not really a bug. And not for vshaxe itself.
But for **vim** + the **nvim-coc** plugin and using the haxe_language_server from vshaxe...

So I have a question.
For the sake of simplicity let's say our project is just:

```
build.hxml
src/Main.hx
```

To launch **vim**, if you do `vim src/Main.hx`, completion works.
If you do, `cd src ; vim Main.hx`, it doesn't. 

The nvim-coc plugins `:CocInfo` reveals it's simply because in the second case, the `build.hxml` is searched in `src/build.hxml`!

Right now my config (as given by `:CocConfig`) contains:
```
{
      ...
     "initializationOptions": {
       "displayArguments": ["build.hxml"]
     },
...
}
```

My question is obvious (or not, I don't know if that's a correct approach): would you guys know a way to have `build.hxml` be looked recursively in parent dirs?  Or is there another way?

In the end, maybe I should just have the discipline, but I guess it doesn't hurt to ask.

**Comment by Gama11** [comment]
Well, from a language server protocol perspective, there is a "workspace directory" that is used as the root, which is what `build.hxml` is resolved against. This maps quite well to editors like VSCode where you always open a specific workspace. I guess Vim doesn't really have such a concept and you only ever open individual files? And nvim-coc just uses the current directory as the "workspace"?

You could possibly do something like this?

```
```

**Comment by emugel** [comment]
Yes no workspaces on vim, it's very common to work on several projects/libs at the same time splitting the screen. Absolute path would work, but just for this project obviously.

So...  
Do you think it's worth asking the author of nvim-coc whether it's possible to have variables/function calls in that config so as to inject a full path their dynamically? Likely it's just a static json file, but I have no idea (there is also a `:CocLocalConfig` but I think it will be too painful to have local config for all projects)

**Comment by emugel** [comment]
Oh, isn't there some magic if you have a `haxelib.json` file where you could define the path for the hxml? I think I have seen that somewhere... At least haxelib.json is standard and not hidden as would be the case for a CocLocalConfig

**Comment by Gama11** [comment]
The only reason you don't need a local config for each project in VSCode is that vshaxe (on the VSCode extension / language _client_ side) auto-detects HXML files in the current workspace. IMO this is not the responsibility of the language server, but the client that manages the language server.

Somebody has already written a [Haxe plugin for coc.nvim](https://github.com/vantreeseba/coc-haxe), I think that the "recursively look for build.hxml" functionality or however you choose to deal with this belongs there (especially since it would be quite VIM-specific).

**Comment by Gama11** [comment]
I'm not sure what "`haxelib.json` magic" you're talking about.

**Comment by emugel** [comment]
> IMO this is not the responsibility of the language server, but the client that manages the language server.
I don't disagree!
Will have a look, a centralized plugin for coc seems ideal indeed. Thank you!

**Comment by emugel** [comment]
For posterity... Found some info here: https://github.com/neoclide/coc.nvim/issues/107

coc.nvim will automatically determines the root, but **only** if there is a `.git/` folder, acting as the root automatically. Or failing that, a `.vim/` folder. 

Meaning if you haven't done a `git init`, the root folder is usually `$HOME`. But it's more configurable in coc.nvim, see `:help g:rooter_patterns`.

So, behaviour I described above (with no autocompletion) only applies to those folders without a `.git/`.
